library/L1_Peripheral/msp432p401r/system_controller.hpp

```diff
@@ -145,111 +226,158 @@ class SystemController final : public sjsu::SystemController
+
+    // =========================================================================
+    // Step 4. Determine the clock rate of each of the primary clocks based on
+    //         the configured clock source and clock dividers.
+    // =========================================================================
+    units::frequency::hertz_t aclk  = 0_Hz;
+    units::frequency::hertz_t mclk  = 0_Hz;
+    units::frequency::hertz_t smclk = 0_Hz;
+    units::frequency::hertz_t bclk  = 0_Hz;
+
+    switch (clock_configuration_.auxiliary.clock_source)
+    {
+      case Oscillator::kLowFrequency: aclk = lfxt; break;
+      case Oscillator::kVeryLowFrequency: aclk = vlo; break;
+      case Oscillator::kReference: aclk = refo; break;
+      default: break;
+    }
 
-  /// Configures the digitally controlled clock (DCO) to produce the desired
-  /// target nominal frequency to drive the master clock.
-  ///
-  /// @param frequency The desired target frequency between 1 MHz and 48 MHz.
-  void SetSystemClockFrequency(
-      units::frequency::megahertz_t frequency) const
-  {
-    const DcoConfiguration_t kDcoConfiguration =
-        CalculateDcoConfiguration(frequency);
+    switch (clock_configuration_.master.clock_source)
+    {
+      case Oscillator::kLowFrequency: mclk = lfxt; break;
+      case Oscillator::kVeryLowFrequency: mclk = vlo; break;
+      case Oscillator::kReference: mclk = refo; break;
+      case Oscillator::kDigitallyControlled: mclk = dco; break;
+      case Oscillator::kModule: mclk = mod; break;
+      case Oscillator::kHighFrequency: mclk = hfxt; break;
+    }
 
-    UnlockClockSystemRegisters();
+    switch (clock_configuration_.subsystem_master.clock_source)
     {
-      clock_system->CTL0 = bit::Insert(clock_system->CTL0,
-                                       kDcoConfiguration.tuning_value,
-                                       Control0Register::kTuningSelect);
-      clock_system->CTL0 = bit::Insert(clock_system->CTL0,
-                                       kDcoConfiguration.frequency_select,
-                                       Control0Register::kFrequencySelect);
-      clock_system->CTL0 =
-          bit::Set(clock_system->CTL0, Control0Register::kEnable);
+      case Oscillator::kLowFrequency: smclk = lfxt; break;
+      case Oscillator::kVeryLowFrequency: smclk = vlo; break;
+      case Oscillator::kReference: smclk = refo; break;
+      case Oscillator::kDigitallyControlled: smclk = dco; break;
+      case Oscillator::kModule: smclk = mod; break;
+      case Oscillator::kHighFrequency: smclk = hfxt; break;
     }
-    LockClockSystemRegisters();
 
-    WaitForClockReadyStatus(Clock::kMaster);
-    SetClockSource(Clock::kMaster, Oscillator::kDigitallyControlled);
-    SetClockSource(Clock::kSubsystemMaster, Oscillator::kDigitallyControlled);
+    switch (clock_configuration_.backup.clock_source)
+    {
+      case Oscillator::kLowFrequency:
+      case Oscillator::kReference:
+      default: break;
+    }
```

Comment:
Whats the purpose of this switch case here?

Reply:
Completely missed this, meant to set `bclk` here.